src/sentry/tasks/data_export.py

```diff
@@ -155,9 +156,71 @@ def attach_eventuser(items):
     return file_name
 
 
-def process_discover(data_export, file):
-    # TODO(Leander): Implement processing for Discover
-    raise NotImplementedError("Discover processing has not been implemented yet")
+def process_discover(data_export, file, limit):
+    """
+    Convert the discovery query to a CSV, writing it to the provided file.
+    Returns the suggested file name.
+    (Adapted from 'src/sentry/api/endpoints/organization_events.py')
+    """
+    payload = data_export.query_info
+
+    start, end = get_date_range_from_params(payload)
+    try:
+        project = Project.objects.get(id=payload["project"])
+    except Project.DoesNotExist as error:
+        metrics.incr("dataexport.error", instance=six.text_type(error))
+        logger.error("dataexport.error: {}".format(six.text_type(error)))
+        raise DataExportError("Requested project does not exist")
+
+    params = {
+        "organization_id": data_export.organization_id,
+        "project_id": [project.id],
+        "start": start,
+        "end": end,
+    }
+
+    def data_fn(offset, limit):
+        return discover.query(
+            selected_columns=payload["field"],
+            query=payload["query"],
+            params=params,
+            offset=offset,
+            limit=limit,
+            referrer="api.organization-events-v2",
+            auto_fields=True,
+            use_aggregate_conditions=True,
+        )
+
+    with snuba_error_handler():
+        # Get a single entry to prepare the header row
+        sample = data_fn(0, 1)["data"]
```

Comment:
You could hit a KeyError here as results don't always have the `data` key when snuba is overloaded or broken.

Reply:
@markstory If snuba is overloaded or broken, won't it throw an error that'll be managed by the `with snuba_error_handler()`? (i.e. SnubaError), or will the KeyError override that?

Reply:
Yes it would. I missed the context manager. My bad.